src/Errata/Internal/Render.hs

```diff
@@ -151,69 +210,69 @@ renderSourceLines slines (Block {..}) lspans = unsplit "\n" sourceLines
                 -- We use the last one as the before-line.
                 (_, _) ->
-                    let n = last es
+                    let (n,l) = last es
                         mid = if
                             | snd (connAround n) -> TB.fromText styleVertical <> " "
                             | hasConnMulti       -> "  "
                             | otherwise          -> ""
-                    in (omitPrefix <> mid) : (linePrefix n <> mid <> showLine [] n) : makeSourceLines 0 ns'
+                    in (omitPrefix <> mid) : (linePrefix n <> mid <> showLine [] l) : makeSourceLines 0 ns'
 
         -- Decorate a line that has pointers.
         -- The pointers we get are assumed to be all on the same line.
-        makeDecoratedLines :: N.NonEmpty Pointer -> [TB.Builder]
-        makeDecoratedLines pointers = (linePrefix line <> TB.fromText lineConnector <> sline) : decorationLines
+        makeDecoratedLines :: [Pointer] -> (Line, source) -> [TB.Builder]
+        makeDecoratedLines pointers (num,line) = 
+            (linePrefix num <> TB.fromText lineConnector <> sline) : decorationLines
             where
                 lineConnector = if
                     | hasConnBefore && hasConnUnder -> styleVertical <> " "
                     | hasConnMulti                  -> "  "
                     | otherwise                     -> ""
 
                 -- Shortcuts to where this line connects to.
-                hasConnHere = hasConn line
-                (hasConnBefore, hasConnAfter) = connAround line
+                hasConnHere = hasConn num
+                (hasConnBefore, hasConnAfter) = connAround num
                 hasConnAround = hasConnBefore || hasConnAfter
                 hasConnOver = hasConnHere || hasConnBefore
                 hasConnUnder = hasConnHere || hasConnAfter
 
                 -- The sorted pointers by column.
                 -- There's a reverse for when we create decorations.
-                pointersSorted = N.fromList . sortOn pointerColumns $ N.toList pointers
-                pointersSorted' = N.reverse pointersSorted
+                pointersSorted = sortOn pointerColumns pointers
+                pointersSorted' = reverse pointersSorted
 
-                -- The line we're on.
-                line = pointerLine $ N.head pointers
-                sline = showLine (map pointerColumns (N.toList pointersSorted)) line
+                -- The line number we're on.
+                sline = showLine (map pointerColumns pointersSorted) line
 
                 -- The resulting decoration lines.
                 decorationLines = if
                     -- There's only one pointer, so no need for more than just an underline and label.
-                    | N.length pointersSorted' == 1                           -> [underline pointersSorted']
+                    | length pointersSorted' == 1                             -> [underline pointersSorted']
```

Comment:
This is pretty big spacing, so I think you should just remove the horizontal alignment here.